cst/plugins/eu.esdihumboldt.cst.functions.geometric/src/eu/esdihumboldt/cst/functions/geometric/GeometryHelperFunctions.groovy

```diff
@@ -150,6 +153,69 @@ class GeometryHelperFunctions {
 	}
 
 	/**
+	 * Specification for a _boundaryCovers function
+	 */
+	public static final HelperFunctionSpecification _boundaryCovers_spec = new HelperFunctionSpecification(
+	"Determine if the boundary of a given geometry covers all points of a line.", "true if the all points of the line are on the geometry's boundary",
+	new HelperFunctionArgument("geometry", "Geometry"),
+	new HelperFunctionArgument("line", "Line"),
+	);
+
+	/**
+	 * Determine if the boundary of a geometry covers all points of a line.
+	 *
+	 * @param args the function arguments
+	 * @return true if the geometry's boundary completely covers the line
+	 */
+	@Nullable
+	static boolean _boundaryCovers(Map args) {
+		return args.geometry.getGeometry().getBoundary().covers(args.line.getGeometry());
+	}
+
+	/**
+	 * Specification for the _spatialIndexQuery function
+	 */
+	public static final HelperFunctionSpecification _spatialIndexQuery_spec = new HelperFunctionSpecification(
+	"Query a spatial index. The bounding box of the given geometry is computed and used to query the spatial index.", "Collection of instances that match the query.",
+	new HelperFunctionArgument("spatialIndex",
+	"The spatial index service to query (e.g. '_spatialIndex'"),
+	new HelperFunctionArgument("geometry",
+	"The geometry whose bounding box is used as a spatial query."));
+
+	/**
+	 * Query a spatial index
+	 *
+	 * @param args the function arguments
+	 * @return a Collection of Instances that match the spatial query
+	 */
+	@Nullable
+	static Collection<Instance> _spatialIndexQuery(Map args) {
+		def result = []
+
+		Geometry geometry = args.geometry.getGeometry()
```

Comment:
I would vote for a consistent behavior w/ the other geometry helper functions regarding what can be passed in for a geometry. For instance that also Instances can be passed in. It should be possible to reuse the functionality for that (as this should already be done in a couple of other places).